### PR TITLE
User input

### PR DESCRIPTION
**Pull request by alwinbaby**
Added User input.py files with the test codes

### CODE REVIEW

**Comment by holtzmak** [comment]
First of all, before I review, it's nice that you tried to pick my commits from the branch I have going, but I want to bring up a better practice: typically when you and another person are working together, you don't share commits. If I change something in my open pull request and you merge in what you have here, I will suddenly have merge conflicts, and vice versa.

In the future, when you make pull requests, it should just contain your commits (unless you needed work that someone else was doing).

File: input.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from typing import Counter
+from unittest.result import failfast
+from spellchecker import SpellChecker
+
+
+def function(sting):
```
Comment (holtzmak) [inline input.py:6]: This should have a name that reflect what it does:
```suggestion
def get_symptoms_from_input(sting):
```

File: input.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from typing import Counter
+from unittest.result import failfast
+from spellchecker import SpellChecker
+
+
+def function(sting):
+    userList = sting.split()
+    Counter=0
```
Comment (holtzmak) [inline input.py:8]: Is a counter some function? The capitalization confuses me, we typically name variables in lowercase (with _ between words)
```suggestion
    counter=0

```

File: input.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    spell=SpellChecker()
+    misspelled=spell.unknown(userList)
+    for word in misspelled:
+        Counter = Counter +1
+    if(Counter!=0):
+        print ("Your input have error in it.... Exiting")
+        return None
+    else: 
+
```
Comment (holtzmak) [inline input.py:17]: Remove needless space. Should we think about installing a python formatter? I made one here https://github.com/holtzmak/Git-Pre-Commit-and-IntelliJ-Settings

File: input.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        print ("Your input have error in it.... Exiting")
+        return None
+    else: 
+        
+        return userList
+
+def get_symptoms_from_user():
+    print("Enter the list of Symptoms by space: ")
+    i =input()
+    while (function(i) == None):
```
Comment (holtzmak) [inline input.py:23]: Maybe the user wouldn't want to re-enter their symptoms. Should we watch for `exit`?

File: test_unittest.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import unittest
+from input import function
+
+class Test(unittest.TestCase):
```
Comment (holtzmak) [inline test_unittest.py:4]: We should probably give this test class a better name:
```suggestion
class TestUserInput(unittest.TestCase):
```
A decent example I found quickly that shows python naming conventions is here https://stackoverflow.com/questions/32672551/python-unittesting-for-a-class-method

File: test_unittest.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import unittest
+from input import function
+
+class Test(unittest.TestCase):
+    def test_correct(self):
```
Comment (holtzmak) [inline test_unittest.py:5]: Similarly, these tests should have names that reflect what they're testing:
```suggestion
    def test_user_input_rejected_containing_mistake(self):
```

File: test_unittest.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import unittest
+from input import function
+
+class Test(unittest.TestCase):
+    def test_correct(self):
+        strin=('feaver cough')
+        i = function(strin)
+        self.assertEqual(None, i)
+    
+    def test_correct2(self):
```
Comment (holtzmak) [inline test_unittest.py:10]:
```suggestion
    def test_user_input_accepted(self):
```

File: test_unittest.py (hunk 5 above)
Comment (holtzmak) [reply test_unittest.py:4]: Similarly, your file names should reflect what things are:

Change `input.py` to something like `user_input_utils.py` and change `test_unittest.py` to `test_user_input.py`

**Review by holtzmak — CHANGES_REQUESTED** [review]
Please address the changes I made in other comments.

I think we should start making folders (or what we call modules in python) for the `src` code and `tests` code. For example, see how code is organized in this repo https://github.com/pallets/flask.

- [x] Make the `src` and `tests` folders/packages/modules and put your two python files in accordingly

File: input.py (hunk 4 above)
Comment (alwinbaby) [reply input.py:24]: Yeah that could work. Also, maybe the user has 3 attempts to get it right. That will be an easy fix.

File: input.py (hunk 3 above)
Comment (alwinbaby) [reply input.py:18]: If space is a problem then we can definitely get rid of it

**Pull request merged**